src/product/ajax.rs

```diff
@@ -4,9 +4,199 @@ use serde::{Deserialize, Deserializer};
 
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TranslationBonusLangs {
+    pub child_count: i32,
+    pub price: i32,
+    pub price_in_tax: i32,
+    pub price_tax: i32,
+    pub recipient_available_count: i32,
+    pub recipient_max: i32,
+    pub status: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(untagged)]
+pub enum TranslationBonusLangsTypes {
+    Arr(Vec<()>),
+    Map(HashMap<String, TranslationBonusLangs>),
+}
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TranslationInfo {
+    pub lang: Option<String>,
+    pub original_workno: Option<String>,
+    pub parent_workno: Option<String>,
+    pub production_trade_price_rate: i32,
+    pub is_volunteer: bool,
+    pub translation_bonus_langs: TranslationBonusLangsTypes,
+    pub is_translation_bonus_child: bool,
+    pub is_translation_agree: bool,
+    pub is_original: bool,
+    pub is_child: bool,
+    pub is_parent: bool,
+    pub child_worknos: Vec<String>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Rank {
+    pub rank_date: String,
+    pub rank: i32,
+    pub category: String,
+    pub term: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct CountDetail {
+    pub count: i32,
+    pub review_point: i32,
+    pub ratio: i32,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct DlCountItem {
+    pub edition_type: String,
+    pub lang: String,
+    pub workno: String,
+    pub display_order: i32,
+    pub display_label: String,
+    pub edition_id: i32,
+    pub label: String,
+    #[serde(deserialize_with = "serde_aux::prelude::deserialize_number_from_string")]
+    pub dl_count: i32,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct VoicePack {
+    pub parent_official_price: Option<i32>,
+    pub child_price: i32,
+    pub parent_price: i32,
+    pub sum_point: i32,
+    pub sum_price: i32,
+    pub product_ids: Vec<String>,
+    pub child_official_price: Option<()>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct SalesEndInfo {
+    pub can_download: bool,
+    pub not_download: bool,
+    pub end_date: HashMap<String, String>,
+    pub end_date_proto: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Bonus {
+    pub end_date_str: String,
+    pub dist_flg: String,
+    pub description: Option<String>,
+    pub title: String,
+    pub end_date: Option<String>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct LimitedFreeTerms {
+    pub workno: String,
+    pub id: i32,
+    pub start_date: String,
+    pub original_workno: Option<String>,
+    pub end_date: String,
+}
+
 /// Data of a product from the AJAX API.
 #[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
```

Comment:
I think request should not be failed even if new field is added. Maybe it is good to add deny_unknown_fields attribute only when testing.

Reply:
yeah makes sense. what do you think about https://docs.rs/serde_ignored/latest/serde_ignored/